spec/action_spec.rb

```diff
@@ -8,6 +8,9 @@ def run(action)
 
   before :each do
     @machine.stub(:id).and_return(EXISTING_UUID)
+    # Vagrant has some pretty buggy multi threading and their conditions
+    # check can fail if the wait_for_ready method returns right away
+    @machine.communicate.stub(:wait_for_ready) { sleep(1) }
```

Comment:
Yes, it should. It was a rather gnarly thing to track down, but Vagrant's condition setting from its threads is such that a race exists if `wait_for_ready` returns right away.

We should probably file a bug against Vagrant, because it is some pretty shoddy code.